src/code/audio/code_8019AF00.c

```diff
@@ -4441,15 +4656,81 @@ void Audio_ClearSariaBgm2(void) {
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A246C.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A2544.s")
+void Audio_PlayMorningSceneSequence(u16 seqId, u8 dayMinusOne) {
+    Audio_StartMorningSceneSequence(seqId);
+    Audio_QueueSeqCmd(0x70000000 | ((u32)(SEQ_PLAYER_BGM_MAIN) << 24) | ((u32)(4) << 0x10) | (u32)(dayMinusOne));
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A257C.s")
+void Audio_StartMorningSceneSequence(u16 seqId) {
+    if (seqId != NA_BGM_AMBIENCE) {
+        Audio_QueueSeqCmd(0x100000FF | ((u8)(SEQ_PLAYER_AMBIENCE) << 24) | ((0) << 16));
+        Audio_StartSceneSequence(seqId);
+        Audio_PlaySequenceWithSeqPlayerIO(SEQ_PLAYER_BGM_MAIN, seqId, 0, 0, 1);
+    } else {
+        Audio_PlayAmbience(AMBIENCE_ID_08);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A25E4.s")
+void Audio_PlaySceneSequence(u16 seqId, u8 dayMinusOne) {
+    if (sRequestedSceneSeqId != seqId) {
+        if (seqId == NA_BGM_AMBIENCE) {
+            Audio_PlayAmbience(AMBIENCE_ID_08);
+        } else if ((seqId != NA_BGM_FINAL_HOURS) || (sPrevMainBgmSeqId == NA_BGM_DISABLED)) {
+            Audio_StartSceneSequence(seqId);
+            Audio_QueueSeqCmd(0x70000000 | ((u32)(SEQ_PLAYER_BGM_MAIN) << 24) | ((u32)(4) << 0x10) |
+                              (u32)(dayMinusOne));
+        }
+        sRequestedSceneSeqId = seqId;
+    }
+}
+
+void Audio_StartSceneSequence(u16 seqId) {
+    u8 fadeInDuration = 0;
+    u8 skipHarpIntro;
+
+    if ((sSeqFlags[sPrevSceneSeqId] & SEQ_FLAG_STORE_PREV_SPOT) &&
+        (sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_STORE_SPOT)) {
+        // Start the sequence from the spot where it last left off last time it was played in the scene
+        if ((sSeqSpot % 64) != 0) {
```

Comment:
Oh, so this is the place to resume? I feel like something to do with it being saved to be resumed would be clearer if this is not used for anything else. Also, this looks like it's carrying a flag rather than actually doing a modulo?

Reply:
So you mean `sSeqSavedSpot`/`SEQ_FLAG_SAVE_SPOT`/`SEQ_FLAG_SAVE_PREV_SPOT` perhaps?

I wasn't sure if `(sSeqSpot & 0x3F) != 0` vs. `(sSeqSpot % 64 != 0`, both work, I can go with the former.

Reply:
Maybe `Position`/`Pos` over "spot", which seems to mean other things in these codebases.

I'm not wrong here, am I, it is a bit strange to do a % 64?

Reply:
` & 0x3F` does make more sense in this context since it's a bit mask instead of a modulo (should have been changed already).

The issue with `Pos` is that it's used way more commonly, especially with sequences that have a world/spatial Position associated with it (think astral observatory where the sequence is placed at a certain position, and gets louder as you get to a certain position).

I haven't PR'd any of those functions yet, but there's two/three separate systems for sequences at certain world positions

Reply:
I can see that, how about `seek`, like `fseek`?